garden/migration.md

```diff
@@ -0,0 +1,465 @@
+- Public headers
+    - e.g. `include/ignition` → `include/gz`
+- Plugins
+    - e.g. `ignition::gazebo::systems::LiftDrag` → `gz::sim::systems::LiftDrag`
+- Shared libraries
+    - e.g. `libignition-gazebo-buoyancy-engine-system.so` → `gz-sim-buoyancy-engine-system`
+- C++ Macros in public headers
+    - e.g. `IGN_PARTITION` → `GZ_PARTITION`
+
+**CMake and Packaging**
+
+- CMake macros/functions
+    - e.g. `ign_find_package()` → `gz_find_package()`
+- CMake macro/function arguments
+    - e.g. `NO_IGNITION_PREFIX` → `NO_PROJECT_PREFIX`
+- CMake variables*
+    - e.g. `IgnOGRE2_FOUND` → `GzOGRE2_FOUND`
+    - Not every CMake variable is tick-tocked, but most of the ones that are used in downstream libraries are
+- CMake packages found by `gz_find_package()`
+    - e.g. `gz_find_package(IgnCURL)` → `gz_find_package(GzCURL)`
+- Debian packages
+    - e.g. `libignition-cmake3-dev` → `libgz-cmake3-dev`
+
+**Misc.**
+
+- Environment variables (names and values)
+    - e.g. `IGN_GAZEBO_RESOURCE_PATH` → `GZ_SIM_RESOURCE_PATH`
+- Command line
+    - e.g. `ign` → `gz`, `ign gazebo` → `gz sim`
+- GUI QML
+    - e.g. `IgnSpinBox` → `GzSpinBox`
+- Topics* (typically in tests)
+    - e.g. `/ignition/XXX` → `/gz/XXX`
+    - **Note:** `/gazebo` is NOT migrated to `/sim`
+- GitHub organizations and repositories
+    - e.g. `ignitionrobotics` → `gazebosim` , `ign-cmake` → `gz-cmake`
+- GitHub actions and workflows
+    - e.g. `ignition-tooling` → `gazebo-tooling`
+- Websites
+    - e.g. `[ignitionrobotics.org](http://ignitionrobotics.org)` → `gazebosim.org`
+- SDF and launch tags
+    - e.g. `<ignition-gui>` → `<gz-gui>`
+
+</details>
+
+### Hard-tocks
+
+There are some exceptions that have been hard-tocked instead, meaning that you **MUST** use the Gazebo counterpart. Using the Ignition counterpart will likely cause compilation or something else to break (unless it’s just a documentation change.)
+
+<details>
+<summary>Hard-tocks</summary>
+
+**Namespaces**
+
+- Ruby namespaces
+    - e.g. `ignition/math` → `gz/math`
+
+**Source**
+
+- C++ Macros in private headers (typically something in a `detail/` directory)
+    - e.g. `DETAIL_IGNITION_UTILS_WARN_RESUME` → `DETAIL_GZ_UTILS_WARN_RESUME`
+- Class names, members, functions, and variables in private headers, or function-local scope
+    - e.g. `kIgnAuthDomain` → `kGzAuthDomain`
+- Install space
+    - e.g. `install/share/ignition` → `install/share/gz`
+- Header guards (where appropriate)
+    - e.g. `IGNITION_COMMON_MATERIAL_HH_` → `GZ_COMMON_MATERIAL_HH_`
+
+**CMake and Packaging**
+
+- Most includeable CMake files in `gz-cmake`
+    - e.g. `IgnUtils.cmake` → `GzUtils.cmake`
+- Gazebo library CMake project names
+    - e.g. `ignition-utils2` → `gz-utils2`
+- Internally used CMake variables
+    - e.g. `IGN_TRANSPORT_VER` → `GZ_TRANSPORT_VER`
+
+**Misc.**
+
+- Gazebo launchfiles and tags
+    - e.g. `sim.ign` → `sim.gzlaunch`, `<ign`  → `<gz`
+- Config and log paths (PENDING)
```

Comment:
Some of the config paths have been tick-tocked. For example, [~/.ignition/gazebo/plugins](https://github.com/gazebosim/gz-sim/blob/ff101099698da7795fe25f9c03b0bc34a7a25f5e/src/SystemLoader.cc#L70-L71)